contracts/scripts/deployRound.ts

```diff
@@ -0,0 +1,143 @@
+
+import { deployMaciFactory } from '../utils/deployment'
+import { MaciParameters } from '../utils/maci'
+
+async function main() {
+  const [deployer] = await ethers.getSigners()
+  let maciFactory = await deployMaciFactory(deployer)
+  await maciFactory.deployTransaction.wait()
+
+  const FundingRoundFactory = await ethers.getContractFactory(
+    'FundingRoundFactory',
+    deployer
+  )
+  const fundingRoundFactory = await FundingRoundFactory.deploy(
+    maciFactory.address
+  )
+  await fundingRoundFactory.deployTransaction.wait()
+
+  const transferOwnershipTx = await maciFactory.transferOwnership(
+    fundingRoundFactory.address
+  )
+  await transferOwnershipTx.wait()
+
+  const userRegistryType = process.env.USER_REGISTRY_TYPE || 'simple'
+  let userRegistry: Contract
+  if (userRegistryType === 'simple') {
+    const SimpleUserRegistry = await ethers.getContractFactory(
+      'SimpleUserRegistry',
+      deployer
+    )
+    userRegistry = await SimpleUserRegistry.deploy()
+    await userRegistry.deployTransaction.wait()
+  } else if (userRegistryType === 'brightid') {
+    const BrightIdUserRegistry = await ethers.getContractFactory(
+      'BrightIdUserRegistry',
+      deployer
+    )
+    // TODO sort out if these arguments are sensible
+    userRegistry = await BrightIdUserRegistry.deploy(
+      '0x636c722e66756e64000000000000000000000000000000000000000000000000', // Do we have a test `context`?
+      fundingRoundFactory.address // Not sure if `verifier` should be EOA or contract account
+    )
+    await userRegistry.deployTransaction.wait()
+  } else {
+    throw new Error('unsupported user registry type')
+  }
+  const setUserRegistryTx = await fundingRoundFactory.setUserRegistry(
+    userRegistry.address
+  )
+  await setUserRegistryTx.wait()
+
+  const recipientRegistryType = process.env.RECIPIENT_REGISTRY_TYPE || 'simple'
+  let recipientRegistry: Contract
+  if (recipientRegistryType === 'simple') {
+    const SimpleRecipientRegistry = await ethers.getContractFactory(
+      'SimpleRecipientRegistry',
+      deployer
+    )
+    recipientRegistry = await SimpleRecipientRegistry.deploy(
+      fundingRoundFactory.address
+    )
+    await recipientRegistry.deployTransaction.wait()
+  } else if (recipientRegistryType === 'optimistic') {
+    const OptimisticRecipientRegistry = await ethers.getContractFactory(
+      'OptimisticRecipientRegistry',
+      deployer
+    )
+    recipientRegistry = await OptimisticRecipientRegistry.deploy(
+      ethers.BigNumber.from(10).pow(ethers.BigNumber.from(18)).div(10),
+      300,
+      fundingRoundFactory.address
+    )
+    await recipientRegistry.deployTransaction.wait()
+  } else {
+    throw new Error('unsupported recipient registry type')
+  }
+
+  const setRecipientRegistryTx = await fundingRoundFactory.setRecipientRegistry(
+    recipientRegistry.address
+  )
+  await setRecipientRegistryTx.wait()
+
+  const rinkebyDaiAddress = '0x5592ec0cfb4dbc12d3ab100b257153436a1f0fea'
```

Comment:
If we're hardcoding the Rinkeby DAI address here, should this just be a `deployRinkebyRound` script?

Reply:
I was thinking about making this a variable in the env instead of hardcoding. Ill refactor this out.